CC | osbuilder: Rust environment cannot be found When `AGENT_SOURCE_BIN` is set

# Description of problem

When AGENT_SOURCE_BIN is set, the Rust environment cannot be found during the attestation agent build process.
```
script -fec 'USE_DOCKER=true AGENT_SOURCE_BIN=/root/go/src/github.com/kata-containers/kata-containers/src/agent/target/x86_64-unknown-linux-musl/release/kata-agent KATA_BUILD_CC=yes AA_KBC="cc_kbc_tdx" ./rootfs.sh "${distro}"'
```

# Expected result

Built rootfs successfull.

# Actual result

```
cd app &&  cargo build --release --no-default-features --features " cc_kbc_tdx ttrpc,rust-crypto" --target x86_64-unknown-linux-gnu
/bin/sh: 1: cargo: not found
make: *** [Makefile:125: build] Error 127
Failed at 693: make KBC=${AA_KBC} ttrpc=true
```

# Further information

(replace this text with any extra information you think might be useful)

--- a/tools/osbuilder/rootfs-builder/rootfs.sh
+++ b/tools/osbuilder/rootfs-builder/rootfs.sh
@@ -606,7 +606,7 @@ EOF
 	AGENT_DIR="${ROOTFS_DIR}/usr/bin"
 	AGENT_DEST="${AGENT_DIR}/${AGENT_BIN}"
 
-	if [ -z "${AGENT_SOURCE_BIN}" ] ; then
+	if [ -z "${AGENT_SOURCE_BIN}" ] || [ -n "${AA_KBC}" ]; then
 		test -r "${HOME}/.cargo/env" && source "${HOME}/.cargo/env"
 		# rust agent needs ${arch}-unknown-linux-${LIBC}
 		if ! (rustup show | grep -v linux-${LIBC} > /dev/null); then
@@ -617,7 +617,9 @@ EOF
 			bash ${script_dir}/../../../ci/install_rust.sh ${RUST_VERSION}
 		fi
 		test -r "${HOME}/.cargo/env" && source "${HOME}/.cargo/env"
+	fi
 
+	if [ -z "${AGENT_SOURCE_BIN}" ]; then
 		agent_dir="${script_dir}/../../../src/agent/"
 
 		if [ "${SECCOMP}" == "yes" ]; then

